packages/matter.js/src/cluster/server/ClusterServer.ts

```diff
@@ -108,7 +110,7 @@ export function ClusterServer<
             return datasource;
         },
 
-        set datasource(newDatasource: ClusterDatasource | undefined) {
+        _setDatasource<S extends Storage>(newDatasource: ClusterDatasource<S> | undefined) {
```

Comment:
Yeah so the ClusterDatasource for the new code is specifically designed be accessed synchronously.  In ClusterBehaviorBacking.invokeInitializer I think we should just assert that it did not return a promise.  Then invokeInitializer wouldn't need to be async.